packages/ui/src/components/pagination/pagination.tsx

```diff
@@ -167,72 +33,89 @@ export const Pagination: FC<PaginationProps> = ({
 
   const totalPages = indeterminate || !totalItems || !pageSize ? undefined : Math.ceil(totalItems / pageSize)
 
-  // Render nothing if `totalPages` is absent or <= 1, and both `nextPage` and `previousPage` are absent
-  if ((!totalPages || totalPages <= 1) && hasNext === undefined && hasPrevious === undefined) {
+  // Render nothing if `totalPages` is absent, and both `nextPage` and `previousPage` are absent
+  if (!totalPages && hasNext === undefined && hasPrevious === undefined) {
     return null
   }
 
+  const handleChangePageSize = (value: number) => {
+    onPageSizeChange?.(value)
+  }
+
+  const renderItemsPerPageBlock = () => {
+    if (!onPageSizeChange) return null
+
+    const options = pageSizeOptions.map(option => ({ label: option, value: option }))
+
+    return (
+      <>
+        <Layout.Horizontal align="center" gap="xs">
+          <Select options={options} value={pageSize} onChange={handleChangePageSize} size="sm" />
+          <Text>items per page</Text>
+        </Layout.Horizontal>
+
+        <Separator orientation="vertical" className="h-8" />
+      </>
+    )
+  }
+
   return (
-    <>
-      <PaginationPrimitive.Root className={cn('mt-cn-xl', className)}>
+    <Layout.Horizontal className={cn('mt-cn-xl w-full', className)} align="center" justify="between">
+      <div>
+        {totalPages && currentPage && (
+          <Layout.Horizontal align="center" gap="md">
+            {renderItemsPerPageBlock()}
+
+            <Text>
+              Page {currentPage} of {totalPages}
+            </Text>
+          </Layout.Horizontal>
+        )}
+      </div>
+
+      <Layout.Horizontal gap="xs">
         {!indeterminate && totalPages && currentPage ? (
-          <PaginationPrimitive.Content
-            className={cn({
-              'cn-pagination-hide-pages': hidePageNumbers
-            })}
-          >
-            {/* Previous Button */}
-            <PaginationPrimitive.Item className="cn-pagination-item-previous">
-              <PaginationPrimitive.Previous
-                onClick={goToPage ? handleGoToPage(currentPage > 1 ? currentPage - 1 : undefined) : undefined}
-                href={getPageLink?.(currentPage > 1 ? currentPage - 1 : currentPage)}
-                disabled={currentPage === 1}
-              />
-            </PaginationPrimitive.Item>
+          <>
+            <PaginationPrimitive.Previous
+              onClick={goToPage ? handleGoToPage(currentPage > 1 ? currentPage - 1 : undefined) : undefined}
+              href={getPageLink?.(currentPage > 1 ? currentPage - 1 : currentPage)}
+              disabled={currentPage === 1}
+            />
 
             {/* Pagination Items */}
```

Comment:
NIT: Not a blocker. Can we add a comment saying:

It is page number logic and later if needed it can be used?